src/jit/codegenxarch.cpp

```diff
@@ -7366,6 +7366,66 @@ void CodeGen::genSSE2BitwiseOp(GenTreePtr treeNode)
     inst_RV_RV(ins, targetReg, operandReg, targetType);
 }
 
+//-----------------------------------------------------------------------------------------
+// genSSE41RoundOp - generate SSE41 code for the given tree as a round operation
+//
+// Arguments:
+//    treeNode  - tree node
+//
+// Return value:
+//    None
+//
+// Assumptions:
+//     i) SSE4.1 is supported by the underlying hardware
+//    ii) tree oper is a GT_INTRINSIC
+//   iii) tree type is floating point type
+//    iv) tree oper is CORINFO_INTRINSIC_Round, _Ceiling, or _Floor
+//     v) caller of this routine needs to call genProduceReg()
+void CodeGen::genSSE41RoundOp(GenTree* treeNode)
+{
+    // i) SSE4.1 is supported by the underlying hardware
+    assert(compiler->compSupports(InstructionSet_SSE41));
+
+    // ii) tree oper is a GT_INTRINSIC
+    assert(treeNode->OperGet() == GT_INTRINSIC);
+
+    GenTree*    srcNode  = treeNode->gtGetOp1();
+
+    instruction ins      = ins_FloatRound(treeNode->TypeGet());
```

Comment:
Using a function it's too much ceremony for something that's simply `(type == TYP_FLOAT) ? INS_roundss : INS_roundsd`. There are already too many cases where codegen uses such functions for no good reason and that results in wasted cycles and wasted time for those who read the code.

Reply:
Will fix. The CORINFO_INTRINSIC_Sqrt function should probably be updated as well (in a separate PR)

Reply:
Fixed.